src/java/org/apache/cassandra/gms/Gossiper.java

```diff
@@ -2416,4 +2421,20 @@ private EndpointState removeEndpointState(InetAddressAndPort endpoint)
         return removedState;
     }
 
+    private void tryTaskLock()
+    {
+        boolean result;
+        try
+        {
+            result = taskLock.tryLock(LOCK_TIMEOUT_MS, TimeUnit.MILLISECONDS);
+        }
+        catch (InterruptedException e)
+        {
+            Thread.currentThread().interrupt();
```

Comment:
out of curiosity - why set this flag?

Reply:
because I don't know the whole Cassandra code :)

Reply:
for example `JVMStabilityInspector` gets changes at some point so that it does something depending on whether the thread was interrupted or not. Why you are so picky about that :D it is a good known way of handling IE